plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/LibvirtStorageAdaptor.java

```diff
@@ -1402,7 +1402,10 @@ public KVMPhysicalDisk copyPhysicalDisk(KVMPhysicalDisk disk, String name, KVMSt
          */
 
         KVMStoragePool srcPool = disk.getPool();
-        PhysicalDiskFormat sourceFormat = disk.getFormat();
+        /* Linstor images are always stored as RAW, but Linstor uses qcow2 in DB,
+           to support snapshots(backuped) as qcow2 files. */
+        PhysicalDiskFormat sourceFormat = srcPool.getType() != StoragePoolType.Linstor ?
+                disk.getFormat() : PhysicalDiskFormat.RAW;
```

Comment:
can set format RAW in KVMPhysicalDisk for Linstor disks?

Reply:
You mean, setting it in `KVMPhysicalDisk.setFormat()` or `KVMPhysicalDisk.getFormat()`?

Reply:
@sureshanaparti ?

Reply:
@sureshanaparti ping

Reply:
@rp- sorry, missed this. if the format is set while creating the linstor disk (KVMStoragePool), you get it with `disk.getFormat()`

Reply:
check in LinstorStorageAdaptor.java - `getPhysicalDisk()` & `createPhysicalDisk()` - where KVMPhysicalDisk obj is created and RAW format is set (so, no need to set the format RAW with StoragePoolType.Linstor type check again).

Reply:
Sadly it is, because the LinstorStorageAdaptor isn't used for migrating from another PrimaryStorage.
And so it will use the format sent from the management server and this doesn't match what is expected from the qemu-img convert command.

Reply:
If you want I can create the migration error and show you the exception/error log.

Reply:
ok, got it. maybe, you can revisit that later.